### PR TITLE
Feature/fix enum empty bug

### PR DESCRIPTION
**Pull request by huhuaaa**
Fixed the bug. The selector have an empty option, when use enum and the default value is 0 or false or ''.

### Reasons for making this change

[Please describe them here]

Fixed the bug. The selector have an empty option, when use enum and the default value is 0 or false or ''.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by epicfaace** [comment]
@huhuaaa Can you explain what issue you are solving? When the default value is 0 or false or "", what do you expect the dropdown to look like?

When the default value is "" or false or "" for me, it looks like this:
image

**Comment by huhuaaa** [comment]
Yes, i use "typeof default === 'undefined'" to check the default value.

**Comment by epicfaace** [comment]
Why would we _not_ want to have an empty option when the default is 0 or false or ''?

**Comment by huhuaaa** [comment]
Yes, when the default is 0 or false or '' , we need an option. But i think an empty option is not the best one.  And there's another bug here，i set enum = [0,1] and enumNames = ['zero', 'one'] , the empty option also showing. This result is not i expected.

[code sample](https://mozilla-services.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJmaXJzdE5hbWUiOmZhbHNlLCJsYXN0TmFtZSI6Ik5vcnJpcyIsImFnZSI6NzUsImJpbyI6IlJvdW5kaG91c2Uga2lja2luZyBhc3NlcyBzaW5jZSAxOTQwIiwicGFzc3dvcmQiOiJub25lZWQiLCJzdGF0dXMiOjB9LCJzY2hlbWEiOnsidGl0bGUiOiJBIHJlZ2lzdHJhdGlvbiBmb3JtIiwiZGVzY3JpcHRpb24iOiJBIHNpbXBsZSBmb3JtIGV4YW1wbGUuIiwidHlwZSI6Im9iamVjdCIsInJlcXVpcmVkIjpbImZpcnN0TmFtZSIsImxhc3ROYW1lIl0sInByb3BlcnRpZXMiOnsic3RhdHVzIjp7InR5cGUiOiJudW1iZXIiLCJ0aXRsZSI6InN0YXR1cyIsImVudW0iOlswLDFdLCJlbnVtTmFtZXMiOlsiemVybyIsIm9uZSJdLCJkZWZhdWx0IjowfX19LCJ1aVNjaGVtYSI6eyJmaXJzdE5hbWUiOnsidWk6YXV0b2ZvY3VzIjp0cnVlLCJ1aTplbXB0eVZhbHVlIjoiIn0sImFnZSI6eyJ1aTp3aWRnZXQiOiJ1cGRvd24iLCJ1aTp0aXRsZSI6IkFnZSBvZiBwZXJzb24iLCJ1aTpkZXNjcmlwdGlvbiI6IihlYXJ0aGlhbiB5ZWFyKSJ9LCJiaW8iOnsidWk6d2lkZ2V0IjoidGV4dGFyZWEifSwicGFzc3dvcmQiOnsidWk6d2lkZ2V0IjoicGFzc3dvcmQiLCJ1aTpoZWxwIjoiSGludDogTWFrZSBpdCBzdHJvbmchIn0sImRhdGUiOnsidWk6d2lkZ2V0IjoiYWx0LWRhdGV0aW1lIn0sInRlbGVwaG9uZSI6eyJ1aTpvcHRpb25zIjp7ImlucHV0VHlwZSI6InRlbCJ9fX19)

image

**Comment by epicfaace** [comment]
Oh, I see, so this is solving #1177

**Comment by epicfaace** [comment]
@huhuaaa Can you add some tests for this?

**Comment by huhuaaa** [comment]
OK

**Comment by huhuaaa** [comment]
I had commited the tests.

File: test/enum_test.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    const { node } = createFormComponent({
+      schema,
+    });
+
+    const options = node.querySelectorAll("option");
+    expect(options[0].innerHTML).eql("0");
+  });
+
+  it("should render a select element and it's first option is 'false', if set the enum and the default value is false.", () => {
```
Comment (epicfaace) [inline test/enum_test.js]: Can you also add a test in which the default value is '', and then it shouldn't render two empty options?

Comment (huhuaaa) [reply test/enum_test.js]: Thank you for your suggestions, i have adjusted it.
